source/globalCommands.py

```diff
@@ -1698,6 +1698,23 @@ def script_braille_toggleTether(self, gesture):
 	script_braille_toggleTether.__doc__ = _("Toggle tethering of braille between the focus and the review position")
 	script_braille_toggleTether.category=SCRCAT_BRAILLE
 
+	def script_braille_toggleFocusContextPresentation(self, gesture):
+		values = [x[0] for x in braille.focusContextPresentations]
+		labels = [x[1] for x in braille.focusContextPresentations]
+		try:
+			index = values.index(config.conf["braille"]["focusContextPresentation"])
+		except:
+			index=0
+		newIndex = index+1 if index+1<len(values) else 0
+		config.conf["braille"]["focusContextPresentation"] = values[newIndex]
+		braille.invalidateCachedFocusAncestors(0)
+		braille.handler.handleGainFocus(api.getFocusObject())
+		# Translators: Reports the new state of braille focus context presentation.

```

Comment:
It might be helpful to note that %s gets replaced with the context presentation setting and give an example:

> ```
> # %s will be replaced with the context presentation setting.
> # For example, the full message might be "Braille focus context presentation Fill display for context changes"
> ```

Actually, given the length of that example, I wonder whether we need a colon here. Normally, we prefer to avoid this for brevity (linguistic correctness is less important for these kinds of things), but this is long enough that it might be confusing without it. Alternatively, we could just cut the prefix altogether and just use the setting name. Arguably, the setting names are long/unique enough that the context will be fairly clear.